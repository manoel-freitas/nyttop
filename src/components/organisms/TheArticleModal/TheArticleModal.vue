<template>
  <NYModalCard
    :activate="true"
    :header-title="article.title"
    @onClose="$emit('onCloseArticle')"
  >
    <NYArticle
      :article="article"
      :image="article.image"
      @onChooseArticle="readArticle"
    />
  </NYModalCard>
</template>

<script>
import NYModalCard from 'molecules/NYModalCard'
import NYArticle from 'molecules/NYArticle'
export default {
  components: {
    NYModalCard,
    NYArticle
  },
  props: {
    article: {
      type: Object,
      require: true,
      default: () => ({})
    }
  },
  methods: {
    readArticle() {
      window.location.href = this.article.url
    }
  }
}
</script>

<style></style>
