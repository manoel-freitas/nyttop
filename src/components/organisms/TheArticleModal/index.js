import TheArticleModal from './TheArticleModal'

export default TheArticleModal
