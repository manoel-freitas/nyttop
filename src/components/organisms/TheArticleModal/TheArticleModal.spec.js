/// <reference types="jest" />

import { mount } from '@vue/test-utils'
import TheArticleModal from './TheArticleModal'

const requiredProps = {
  article: {
    image: {
      url: '',
      caption: '',
      width: '',
      heigth: ''
    }
  }
}

const theArticleModal = mount(TheArticleModal, {
  propsData: requiredProps
})
test('TheArticleModal has to be mounted', () => {
  expect(theArticleModal.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(theArticleModal.element).toMatchSnapshot()
})
