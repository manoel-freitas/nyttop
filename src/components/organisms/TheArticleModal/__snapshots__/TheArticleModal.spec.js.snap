// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <section
      class="modal-card-body"
    >
      <button
        aria-label="close"
        class="delete is-pulled-right"
      />
       
      <article
        class="media"
      >
        <div
          class="media-content"
        >
          <div
            class="content"
          >
            <figure
              class="image"
            >
              <img
                alt=""
                src=""
                width=""
              />
            </figure>
             
            <p>
              <strong
                style="max-width: 50px;"
              >
                
          
        
              </strong>
               
              <br />
               
              <small
                class="has-text-grey is-size-7"
              >
                
          
          
        
              </small>
               
              <br />
               
              <span
                class="is-size-6"
              >
                
              </span>
            </p>
             
            <!---->
          </div>
        </div>
      </article>
    </section>
  </div>
</div>
`;
